### PR TITLE
feat(platform): search field add missing examples and refresh mode

### PR DESCRIPTION
**Pull request by genereg1**
## Related Issue.
Closes https://github.com/SAP/fundamental-ngx/issues/4233
Closes https://github.com/sap/fundamental-ngx/issues/4237

## Description

- added Search Field Mobile example 
- added Refresh button behaviour 

##### During Implementation
1. Visual Testing:
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [x] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

##### PR Review
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [x] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [x] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [x] API boundary values
- [x] different combinations of components - free style
- [x] change the API values during testing
3. Documentation and Example validations
- [x] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

### CODE REVIEW

File: apps/docs/src/app/platform/component-docs/platform-search-field/platform-search-field-examples/platform-search-field-mobile/platform-search-field-mobile-example.component.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    suggestions: SuggestionItem[];
+
+    mobileConfig: MobileModeConfig = {
+        approveButtonText: 'OK',
+        hasCloseButton: true,
+    };
+
+    isLoading = true;
+
+    ngOnInit(): void {
```
Comment (Betrozov) [inline apps/docs/src/app/platform/component-docs/platform-search-field/platform-search-field-examples/platform-search-field-mobile/platform-search-field-mobile-example.component.ts:23]: you do not need ngOnInit
please remove it

File: libs/platform/src/lib/search-field/search-field-mobile/search-field-mobile.component.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    selector: 'fdp-search-field-mobile',
+    templateUrl: 'search-field-mobile.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchFieldMobileComponent extends MobileModeBase<SearchFieldMobileInterface> implements OnInit, OnDestroy {
+    childContent: {
+        inputFieldTemplate: TemplateRef<any>,
+        suggestionMenuTemplate: TemplateRef<any>
+    } = null;
```
Comment (Betrozov) [inline libs/platform/src/lib/search-field/search-field-mobile/search-field-mobile.component.ts:32]: please create interface for it

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 3]
```diff
@@ -511,7 +603,23 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
...
+                    if (this.refreshId === target.id) {
+                        return;
+                    }
+
+                    if (this.clearId === target.id) {
+                        if (this.isSearchDone) {
+                            return;
+                        }
+                        this._isFocused = false;
+                    }
```
Comment (Betrozov) [inline libs/platform/src/lib/search-field/search-field.component.ts:621]:
```suggestion
                    if (this.clearId === target.id && !this.isSearchDone) {
                        this._isFocused = false;
                    }
```

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/search-field/search-field-mobile/search-field-mobile.component.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    constructor(
+        @Inject(SEARCH_FIELD_COMPONENT) searchFieldComponent: SearchFieldMobileInterface,
+        elementRef: ElementRef,
+        dialogService: DialogService,
+        @Optional() @Inject(MOBILE_MODE_CONFIG) mobileModes: MobileModeConfigToken[]
+    ) {
+        super(elementRef, dialogService, searchFieldComponent, MobileModeControl.SEARCH_FIELD, mobileModes)
+    }
+
+    ngOnInit(): void {
```
Comment (platon-rov) [inline libs/platform/src/lib/search-field/search-field-mobile/search-field-mobile.component.ts:47]: TypeDoc.

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 5]
```diff
@@ -201,6 +213,9 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
     @Output()
     cancelSearch: EventEmitter<void> = new EventEmitter();
 
+    @Output()
```
Comment (platon-rov) [inline libs/platform/src/lib/search-field/search-field.component.ts:216]: TypeDoc.

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 6]
```diff
@@ -239,13 +254,20 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
...
+    isOpen = false;
+
+    isRefresh = true;
+
     private _currentSearchSuggestionAnnoucementMessage = '';
     private _suggestionOverlayRef: OverlayRef;
     private _suggestionPortal: TemplatePortal;
     private _suggestionkeyManager: FocusKeyManager<SearchFieldSuggestionDirective>;
     private _isFocused = false;
+    isSearchDone = false;
```
Comment (platon-rov) [inline libs/platform/src/lib/search-field/search-field.component.ts:270]: Let's organise fields like `public` then `private`.

**Review by platon-rov — CHANGES_REQUESTED** [review]
1. Dialog doesn't open on input focus.
2. In dialog focus trapped on the input, cannot go anywhere.
3. Dialog doesn't close with `Escape`.
4. Refresh button disappears when input in focus, so cannot tab into it.
5. Cannot tab into any buttons in the input like Clear, Search.
6. When having option select in dialog, then pressing Refresh the Search button disappears.

File: apps/docs/src/app/platform/component-docs/platform-search-field/platform-search-field.module.ts [hunk 7]
```diff
@@ -30,7 +33,9 @@ const routes: Routes = [
         PlatformSearchFieldHeaderComponent,
         PlatformSearchFieldBasicExampleComponent,
         PlatformSearchFieldCategoriesExampleComponent,
-        PlatformSearchFieldDataSourceExampleComponent
-    ]
+        PlatformSearchFieldDataSourceExampleComponent,
+        PlatformSearchFieldMobileExampleComponent
+    ],
+    providers: [{ provide: MOBILE_MODE_CONFIG, useValue: SEARCH_FIELD_MOBILE_CONFIG, multi: true }],
```
Comment (platon-rov) [inline apps/docs/src/app/platform/component-docs/platform-search-field/platform-search-field.module.ts:39]: We're not providing config in the components' modules. It should be provided in Docs modules.

**Review by platon-rov — CHANGES_REQUESTED** [review]
It should probably be `feat(platform)`

File: libs/platform/src/lib/search-field/search-field.component.ts [hunk 8]
```diff
@@ -78,6 +82,14 @@ export class SearchFieldComponent extends BaseComponent implements OnInit, OnDes
     @Input()
```
Comment (platon-rov) [inline libs/platform/src/lib/search-field/search-field.component.ts:82]: Let make this component implementing `SearchFieldMobileInterface`.

File: libs/platform/src/lib/search-field/public_api.ts [hunk 9]
```diff
@@ -1,2 +1,3 @@
 export * from './search-field.component';
 export * from './search-field.module';
+export * from './search-field-mobile/search-field-mobile.module';
```
Comment (platon-rov) [inline libs/platform/src/lib/search-field/public_api.ts:3]: Please also export `search-field-mobile.interface.ts`.

**Review by platon-rov — CHANGES_REQUESTED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Pull request merged**